core/naive-strategy.go

```diff
@@ -201,6 +203,126 @@ func (st *NaiveStrategy) UnrelayedPackets(ctx context.Context, src, dst *Provabl
 	}, nil
 }
 
+func (st *NaiveStrategy) ProcessTimeoutPackets(ctx context.Context, src, dst *ProvableChain, sh SyncHeaders, rp *RelayPackets) error {
+	logger := GetChannelPairLogger(src, dst)
+	var (
+		srcPackets   PacketInfoList
+		dstPackets   PacketInfoList
+		srcLatestHeight             ibcexported.Height
+		srcLatestTimestamp          uint64
+		srcLatestFinalizedHeight    ibcexported.Height
+		srcLatestFinalizedTimestamp uint64
+		dstLatestHeight             ibcexported.Height
+		dstLatestTimestamp          uint64
+		dstLatestFinalizedHeight    ibcexported.Height
+		dstLatestFinalizedTimestamp uint64
+	)
+
+	if 0 < len(rp.Src) {
+		if h, err := dst.LatestHeight(ctx); err != nil {
+			logger.Error("fail to get dst.LatestHeight", err)
+			return err
+		} else {
+			dstLatestHeight = h
+		}
+
+		if t, err := dst.Timestamp(ctx, dstLatestHeight); err != nil {
+			logger.Error("fail to get dst.Timestamp of latestHeight", err)
+			return err
+		} else {
+			dstLatestTimestamp = uint64(t.UnixNano())
+		}
+
+		dstLatestFinalizedHeight = sh.GetLatestFinalizedHeader(dst.ChainID()).GetHeight()
+		if t, err := dst.Timestamp(ctx, dstLatestFinalizedHeight); err != nil {
+			logger.Error("fail to get dst.Timestamp of latestFinalizedHeight", err)
+			return err
+		} else {
+			dstLatestFinalizedTimestamp = uint64(t.UnixNano())
+		}
+	}
+	if 0 < len(rp.Dst) {
+		if h, err := src.LatestHeight(ctx); err != nil {
+			logger.Error("fail to get src.LatestHeight", err)
+			return err
+		} else {
+			srcLatestHeight = h
+		}
+		if t, err := src.Timestamp(ctx, srcLatestHeight); err != nil {
+			logger.Error("fail to get src.Timestamp", err)
```

Comment:
Please modify the error messages on L252 and L260 so that they can be distinguished from each other.